packages/flutter/lib/src/painting/borders.dart

```diff
@@ -294,6 +299,26 @@ class BorderSide {
     );
   }
 
+  /// Get the amount of the stroke width that lies inside of the [BorderSide].
+  ///
+  /// For example, this will return the [width] for a [strokeAlign] of -1, half
+  /// the [width] for a [strokeAlign] of 0, and 0 for a [strokeAlign] of 1.
+  double get strokeInset => width * (1 - (1 + strokeAlign) / 2);
+
+  /// Get the amount of the stroke width that lies outside of the [BorderSide].
+  ///
+  /// For example, this will return 0 for a [strokeAlign] of -1, half the
+  /// [width] for a [strokeAlign] of 0, and the [width] for a [strokeAlign]
+  /// of 1.
+  double get strokeOutset => width * (1 + strokeAlign) / 2;
+
+  /// The offset of the stroke, taking into account the stroke alignment.
```

Comment:
Isn't it half the offset?

Incidentally, as a public API, it's not quite clear why this is a half offset, or why I'd need that.  I think it might be better to just call it `strokeOffset` and then divide it by two where you use it: that value is more interesting. Yes, it's more typing in your code, but I think it makes for a more useful, clear API.

I mean, really, there's not really a huge need for this accessor anyhow, it's just convenient for implementing borders.